File: src/main/java/com/logosprog/mainutils/gui/animation/AnimationListener.java
/*
 * Copyright (c) 2015. This code is a LogosProg property. All Rights Reserved.
 */

package com.logosprog.mainutils.gui.animation;

/**
 * Created by forando on 23.06.15.<br/>
 * Specifies basic animation events.
 */
public interface AnimationListener<T> {
    void onAnimationStart(T obj);
    void onAnimationEnd(T obj);
    void reDraw();
}

File: src/main/kotlin/com/logosprog/mainutils/gui/animation/AnimationListener.kt
/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

@file:Suppress("unused")

package com.logosprog.mainutils.gui.animation

/**
 * Specifies basic com.logosprog.display.animation events.
 */
interface AnimationListener<T> {
    fun onAnimationStart(obj: T?)
    fun onAnimationEnd(obj: T?)
    fun reDraw()
}